src/Codeception/Lib/ModuleContainer.php

```diff
@@ -246,18 +429,79 @@ protected function getModuleConfig($module)
             return $config;
         }
 
-
-        // get config for enabled modules
         foreach ($this->config['modules']['enabled'] as $enabledModuleConfig) {
             if (!is_array($enabledModuleConfig)) {
                 continue;
             }
+
             $enabledModuleName = key($enabledModuleConfig);
-            if ($enabledModuleName !== $module) {
-                continue;
+            if ($enabledModuleName === $moduleName) {
+                return Configuration::mergeConfigs(reset($enabledModuleConfig), $config);
             }
-            $config = Configuration::mergeConfigs(reset($enabledModuleConfig), $config);
         }
+
         return $config;
     }
+
+    /**
+     * Check if there are conflicting modules in this ModuleContainer.
+     *
+     * @throws \Codeception\Exception\ModuleConflictException
+     */
+    public function validateConflicts()
+    {
+        $canConflict = [];
+        foreach ($this->modules as $moduleName => $module) {
+            $parted = $module instanceof PartedModule && $module->_getConfig('part');
+
+            if ($this->active[$moduleName] && !$parted) {
+                $canConflict[] = $module;
+            }
+        }
+
+        foreach ($canConflict as $module) {
+            foreach ($canConflict as $otherModule) {
+                $this->validateConflict($module, $otherModule);
+            }
+        }
+    }
+
+    /**
+     * Check if the modules passed as arguments to this method conflict with each other.
+     *
+     * @param \Codception\Module $module
+     * @param \Codception\Module $otherModule
+     * @throws \Codeception\Exception\ModuleConflictException
+     */
+    private function validateConflict($module, $otherModule)
+    {
+        if ($module === $otherModule || !$module instanceof ConflictsWithModule) {
+            return;
+        }
+
+        $conflicts = $this->normalizeConflictSpecification($module->_conflicts());
+        if ($otherModule instanceof $conflicts) {
+            throw new ModuleConflictException($module, $otherModule);
+        }
+    }
+
+    /**
+     * Normalize the return value of ConflictsWithModule::_conflicts() to a class name.
+     * This is necessary because it can return a module name instead of the name of a class or interface.
```

Comment:
I added it because of this test: https://github.com/Codeception/Codeception/blob/2.2/tests/unit/Codeception/Lib/ModuleContainerTest.php#L168